enterprise/backend/src/metabase_enterprise/caching/api.clj

```diff
@@ -0,0 +1,118 @@
+(ns metabase-enterprise.caching.api
+  (:require
+   [compojure.core :refer [GET]]
+   [metabase.api.common :as api]
+   [metabase.api.common.validation :as validation]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.events :as events]
+   [metabase.util :as u]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(defn- assert-valid-models [model ids]
+  (if (= model "root")
+    (when-not (zero? (first ids))
+      (throw (ex-info (tru "Root configuration is only valid with model_id = 0") {:status-code 400
+                                                                                  :model_id    (first ids)})))
+    (api/check-404 (t2/select-one (case model
+                                    "database"  :model/Database
+                                    "dashboard" :model/Dashboard
+                                    "question"  :model/Card)
+                                  :id [:in ids]))))
+
+(defn- audit-caching-change! [id target prev new]
+  (events/publish-event!
+   :event/caching-update
+   {:user-id api/*current-user-id*
+    :model   :model/CacheConfig
```

Comment:
Uhhh I've written this code without thinking much it seems. Now I'm really worried that `:model-id` and `:details -> :model-id` are going to be confusing. What do you think about `target + target-id` inside details to point to the original model?

Reply:
Yeah I like using target. Or `:target-model` even.

In practice I'm not sure it'll be that confusing though, there are a bunch of other events that use `model-id` for both, with different meanings. For example CSV uploads uses the underlying table as the top -evel model, and the "model" we create on top of it as the `:details` model.

Reply:
For posterity, I see you went with just overloading what model means in the top-level map and the `:details` map rather than introducing the `target` terminology. LGTM, I don't think there will be any confusion in practice and its nice to avoid the need for more domain language.

Reply:
Ah, sorry, missed writing a comment here. Yep, it seemed to me that previous art indicates it's fine to use `model` terminology at both levels, so I just did not want to introduce more terminology.